### PR TITLE
Do not use datalists for autocompletion

### PR DESCRIPTION
**Pull request by Tristramg**

### CODE REVIEW

File: app/templates/components/station-autocomplete.hbs [hunk 1]
```diff
@@ -1,8 +1,5 @@
-<input id='autocomplete-input' list="stations-list" class="form-control" placeholder="Search station…" />
 {{#if stations.isLoaded}}
-<datalist id='stations-list' >
-  {{#each stations.all as |station|}}
-  <option value="{{station.id}}">{{station.name}} ({{station.id}})</option>
-  {{/each}}
-</datalist>
+{{#power-select search=(action "findStation") searchField="name" selected=selected onchange=(action "displayStation") as |station|}}
+{{#if station.is_suggestable}}<i class="fa fa-check"></i>{{/if}}{{station.name}} ({{station.id}})
```
Comment (Signez) [inline app/templates/components/station-autocomplete.hbs]: I would have added a space between `{{/if}}` and `{{station.name}}`, and added a `fa-fw` class to the icon for better rendering.

File: app/templates/components/station-autocomplete.hbs [hunk 2]
```diff
@@ -1,8 +1,5 @@
-<input id='autocomplete-input' list="stations-list" class="form-control" placeholder="Search station…" />
 {{#if stations.isLoaded}}
-<datalist id='stations-list' >
-  {{#each stations.all as |station|}}
-  <option value="{{station.id}}">{{station.name}} ({{station.id}})</option>
-  {{/each}}
-</datalist>
+{{#power-select search=(action "findStation") searchField="name" selected=selected onchange=(action "displayStation") as |station|}}
```
Comment (Signez) [inline app/templates/components/station-autocomplete.hbs]: Does it still work if you break that line, after `selected`?

It would be clearer IMHO.

**Comment by kemenaran** [comment]
Just by curiosity, what was the issue with datalists that motivated this change ?

File: app/components/station-autocomplete.js [hunk 3]
```diff
@@ -4,10 +4,23 @@ import service from 'ember-service/inject';
...
 
-  didInsertElement() {
-    this.$('#autocomplete-input').on('select', (evt) => {
-      this.sendAction('displayStation', evt.target.value);
-      evt.target.value = '';
-    });
+  actions: {
+    findStation(term) {
+      if(term.length > 2) {
+        return this.get('stations.all').filter((station) => {
```
Comment (kemenaran) [inline app/components/station-autocomplete.js]: J'ai l'impression qu'un _early-return_ serait pertinent ici :

```
if (term.length > 2) {
  return [];
}

return this.get('stations.all').filter…
```

**Comment by kemenaran** [comment]
Sinon LGTM

**Comment by Tristramg** [comment]
Updated! Merge?

**Pull request merged**